### PR TITLE
Fix Makefile

### PR DESCRIPTION
**Pull request by cblecker**
Fixing a bunch of the boilerplate Makefile stuff to be less complex. This also fixes compilation on darwin, because go is smart enough to always use the current GOOS/GOARCH, so we don't need to hard code this to linux.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,26 +16,25 @@ fmt:
...
+	go generate ${BUILDFLAGS} ./...
 	@git diff --exit-code -- ./pkg/provider/aws/mock
 
 ensure-mockgen:
-	GO111MODULE=off go get -u github.com/golang/mock/mockgen
+	go get github.com/golang/mock/mockgen@v1.4.4
 
 test:
-	$(GO) test ${BUILDFLAGS} ./... -cover $(PACKAGES)
+	$(GO) test ${BUILDFLAGS} ./... -covermode=atomic -coverpkg=./...
```
Comment (iamkirkbater) [inline Makefile:40]: I'm getting the following error with this:

```> make test
test  ./... -covermode=atomic -coverpkg=./...
/bin/sh: line 0: test: -covermode=atomic: binary operator expected
make: *** [test] Error 2
```

Comment (iamkirkbater) [reply Makefile:40]: Ahh, 

```suggestion
	go test ${BUILDFLAGS} ./... -covermode=atomic -coverpkg=./...
```

**Review by iamkirkbater — CHANGES_REQUESTED** [review]
One small change, other than that it lgtm.

**Review by iamkirkbater — APPROVED** [review]

**Review by clcollins — COMMENTED** [review]
/lgtm

**Pull request merged**